initial pull request for code review

SviatAQA opened the pull request.

SviatAQA committed:
initial pull request for code review

pmichalskiGD commented on mvnw:
```diff
@@ -0,0 +1,316 @@
+#!/bin/sh
+# ----------------------------------------------------------------------------
+# Licensed to the Apache Software Foundation (ASF) under one
```
do we need this file in repository?

SviatAQA replied:
Hello,
File was deleted from the repository and added to .gitignore

pmichalskiGD commented on mvnw.cmd:
```diff
@@ -0,0 +1,188 @@
+@REM ----------------------------------------------------------------------------
```
do we need this file in repository?

SviatAQA replied:
Hello,
File was deleted from the repository and added to .gitignore

pmichalskiGD commented on phonebook.iml:
```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
You should never push iml files into repository

SviatAQA replied:
Hello,
File was deleted from the repository and added to .gitignore

pmichalskiGD commented on pom.xml:
```diff
@@ -0,0 +1,73 @@
+			<scope>provided</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<version>RELEASE</version>
+			<scope>test</scope>
+		</dependency>
```
this entry is duplicated

SviatAQA replied:
Fixed, My bad =(

pmichalskiGD commented on src/main/java/springcourse/phonebook/model/UserContact.java:
```diff
@@ -0,0 +1,38 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String contactName;
+    private String phoneNumber;
```
Shouldn't we keep list of phone numbers?
In task description it's specified that: 'PUT/{name}; phoneNumber - add phone to existing name' so I assume one contact can have several phone numbers

SviatAQA replied:
Fixed.
I implemented it as a List of phone numbers.

pmichalskiGD commented on src/main/java/springcourse/phonebook/model/UserContact.java:
```diff
@@ -0,0 +1,38 @@
+    private String contactName;
+    private String phoneNumber;
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getContactName() {
+        return contactName;
+    }
+
+    public void setContactName(String firstName) {
+        this.contactName = firstName;
+    }
+
+    public String getPhoneNumber() {
+        return phoneNumber;
+    }
+
+    public void setPhoneNumber(String lastName) {
+        this.phoneNumber = lastName;
+    }
```
I highly encourage to use lombok: https://projectlombok.org/

SviatAQA replied:
Fixed.
Lombok library is added.

pmichalskiGD commented on src/main/resources/messages.properties:
```diff
@@ -0,0 +1,5 @@
+#labels
```
are those properties used anywhere?

SviatAQA replied:
Fixed
File was removed

pmichalskiGD reviewed (commented):
I need to say that I'm very pleased with overall code quality. Everything is well formatted and organized, and we only have couple of not needed files in this PR.

I found couple of bugs in the implementation:
- only one phone number per contact (also covered in comments)
- you can send 'id' in create contact request, and this allows to overwrite existing contact
- if you have contact with the same name, you will get 500 when you try to change phone number or delete it
- it's possible to set complete rubbish as phone number. It would be good to introduce some basic validation

SviatAQA committed:
-removed duplicated dependency
-added redundant files to gitignore
-added lombok library and connected it to UserContact class

SviatAQA committed:
- changed phoneNumbers to List
- added exception and verification for unique contacts

SviatAQA committed:
- changed phoneNumbers to List
- added exception and verification for unique contacts

SviatAQA committed:
id value fix / basic validation

SviatAQA committed:
Delete mvnw

SviatAQA committed:
Delete mvnw.cmd

SviatAQA committed:
Delete phonebook.iml

pmichalskiGD reviewed (approved):
Looks fine, ready to be merged.

Thank you for addressing all the comments!
